### ISSUE TITLE
[ux] Native tab focus must not be visually supressed

### ISSUE THREAD

**Issue by mvorisek** [fomantic/Fomantic-UI#2799]
# Feature Request

When `tabindex="0"` is set to a (div) button, the div can be focused by a tab.

Currently when such div is focused, no native glow is present. I have identified this is because:
![image](https://github.com/fomantic/Fomantic-UI/assets/2228672/337be112-e534-48fa-bfc5-6a85906f29cb)
`outline: none` style.

## Example (if possible)
1. open https://dev.atk4.org/demos/form/form.php
2. click the first input field
3. press tab
4. notice the button background changed very slightly, but native tab outline around the button is not present

## Screenshot (if possible)

normal:
![image](https://github.com/fomantic/Fomantic-UI/assets/2228672/bcfe767a-edf6-4de1-8f41-cb055a6c5de9)

focused:
![image](https://github.com/fomantic/Fomantic-UI/assets/2228672/fd4d8e95-8051-4511-adcd-a2434bc02b2f)

compare it with tab glow in https://dev.atk4.org/demos/interactive/paginator.php for example:
![image](https://github.com/fomantic/Fomantic-UI/assets/2228672/ce7db4f3-f38f-4037-ad90-5ea4b78bbaaa)

# Expected fix/behaviour

Not sure why there is `outline: none`, but at least for `:focus-visible` pseudo class no `outline: none` style must be applied.

![image](https://github.com/fomantic/Fomantic-UI/assets/2228672/0ab744e8-06c5-4096-a395-33db3f3f27a5)
(focused button without `outline: none` style, notice the native outline/border around the button, the color is not much different visually, but the visual dimension difference makes it very easy to understand what button is (tab) focused)

**Comment by ko2in** [comment]
Yes. As far as I know, it's by design ever since we've forked from SUI.

This [less](https://github.com/fomantic/Fomantic-UI/blob/develop/src/themes/default/elements/button.overrides) file is your friend for any customization where you can add the outline that suits your design and color pattern.

**Comment by mvorisek** [comment]
This is UX problem of Fomantic-UI. It is present also with dropdown when navigated via tab.

regex to find empty outlines: `outline(-\w+)*:.*(non|0)`

local fix can be:
```
*:focus-visible {
  outline: 2px solid darkblue !important;
}
```

but this is not optimal, as it will override every element, even the ones that Fomantic-UI styles now like inputs.

**Comment by ZoeBijl** [comment]
For what it’s worth, I agree with @mvorisek: this is a bug. Visible focus outline is an important interaction affordance; one especially helpful for power users and accessibility. Turning this off by default deteriorates the functionality of the design. Further more it means any product built on top of Fomantic now fails [$2.4.7](https://www.w3.org/TR/WCAG22/#focus-visible) of [WCAG 2.2](https://www.w3.org/TR/WCAG22/); the international standard for accessible web content. 

Since WCAG is included in the EU’s [EN 301 549: Accessibility requirements for ICT products and services](https://accessible-eu-centre.ec.europa.eu/content-corner/digital-library/en-3015492021-accessibility-requirements-ict-products-and-services_en) any product built on top of Fomantic now fails the EU laws by default. That seems like an undesirable outcome.

Any reason not to drop this into a breaking change update?

**Comment by lubber-de** [comment]
As the next release will be a breaking change, we can enhance the existing draft pr #2801 (and use focus-visible there then) accordingly and consider this getting implemented .
As I mentioned , it will need some effort for special variants to look good cross browser

**Comment by ZoeBijl** [comment]
Would you like any help with the focus style? I’m working on a [change for Forgejo](https://codeberg.org/forgejo/forgejo/pulls/7782) (which is built on Fomantic) to implement better focus styles. Let me know if/what I can do :)

**Comment by lubber-de** [comment]
> Would you like any help with the focus style? I’m working on a [change for Forgejo](https://codeberg.org/forgejo/forgejo/pulls/7782) (which is built on Fomantic) to implement better focus styles. Let me know if/what I can do :)

Sure, any help is appreciated! Just make sure any possible focussable element which supports `outline` on `focus-visible` is covered.
And, it should be cross browser compatible having the same look (outline-style/width/color/offset) on edge/chrome/firefox/safari

You may create try a a fork of [this jsfiddle](https://jsfiddle.net/hpetaxy0) where nearly all fui elements are displayed (initially for theming purposes) and adjust some `tabindex="0"` attributes to non input elements to test this more deeply

Also prepare a form inside a modal (which has focus trap)

**Comment by lubber-de** [comment]
FYI: My PR #2801 was now updated to be able to style (or revert) outline on focus-visible
